src/pgduckdb_duckdb.cpp

```diff
@@ -301,28 +323,30 @@ DuckDBManager::CreateConnection() {
 	return connection;
 }
 
-static bool transaction_handler_configured = false;
-bool started_duckdb_transaction = false;
-
 /* Returns the cached connection to the global DuckDB instance. */
 duckdb::Connection *
-DuckDBManager::GetConnection() {
+DuckDBManager::GetConnection(bool force_transaction) {
 	if (!pgduckdb::IsDuckdbExecutionAllowed()) {
 		elog(ERROR, "DuckDB execution is not allowed because you have not been granted the duckdb.postgres_role");
 	}
 
 	auto &instance = Get();
 	auto &context = *instance.connection->context;
 
-	if (!transaction_handler_configured) {
-		RegisterDuckdbXactCallback();
+	if (!context.transaction.HasActiveTransaction()) {
+		if (IsSubTransaction()) {
+			elog(ERROR, "SAVEPOINT is not supported in DuckDB");
```

Comment:
We probably want to throw in this file no? (I see that L330 we're already `elog(ERROR` though, so maybe something to change later)

Also regarding the error message, is it what PG user would expect? More than something less technical like "Nested transactions are not supported in DuckDB"?